### PR TITLE
EPMRPP-33461 failed cases non passed test cases trend charts

### PR DESCRIPTION
**Pull request by norbert-csaba-herczeg-epam**

### CODE REVIEW

File: app/src/components/widgets/charts/failedCasesTrendChart/configuration.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const range = (start = 0, stop, step = 1) => {
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/failedCasesTrendChart/configuration.js]: This should be moved to utils.
This function needs tests. Here's my idea for a set of tests. Feel free to extend it (I think edge cases should be covered too, for example, range below zero, etc...).
```
describe('range', () => {
  test('it should return an array');
  test('it should return an empty array when invoked with no arguments');
  test('it should return an array with correct length');
  test('it should return an array of numbers increasing on step');
});
```

File: app/src/components/widgets/charts/failedCasesTrendChart/configuration.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  const arr = new Array(length);
+
+  while (idx < length) {
+    arr[idx++] = start; // eslint-disable-line
+    start += step; // eslint-disable-line
+  }
+
+  return arr;
+};
+const rangeMaxValue = (itemsLength) => (itemsLength > 6 ? Math.round(itemsLength / 12) : 1);
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/failedCasesTrendChart/configuration.js]: What is 6? What is 12?
Move magic numbers to constants with meaningful names.

File: app/src/components/widgets/charts/failedCasesTrendChart/failedCasesTooltipContent.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { dateFormat } from 'common/utils/timeDateUtils';
+import styles from '../common/tooltip/tooltipContent/tooltipContent.scss';
```
[the remark this answers is not in the document]
Comment (norbert-csaba-herczeg-epam) [reply app/src/components/widgets/charts/failedCasesTrendChart/failedCasesTooltipContent.jsx]: @DiscoElevator I believe all of the tooltips should go trough a major refactor once all of the components are implemented. I wouldn't block any chart component implementation separately with this task. When all is done we could go trough all of them and conclude with the best way to refactor them.

File: app/src/components/widgets/charts/failedCasesTrendChart/failedCasesTrendChart.stories.jsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Report Portal.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { storiesOf } from '@storybook/react';
+import { host } from 'storybook-host';
+import { withReadme } from 'storybook-readme';
+import 'c3/c3.css';
```
[the remark this answers is not in the document]
Comment (norbert-csaba-herczeg-epam) [reply app/src/components/widgets/charts/failedCasesTrendChart/failedCasesTrendChart.stories.jsx]: @DiscoElevator since the common C3Chart components has been introduced, it is not any more. Thanks! :)

File: app/src/components/widgets/charts/failedCasesTrendChart/configuration.js (hunk 2 above)
Comment (norbert-csaba-herczeg-epam) [reply app/src/components/widgets/charts/failedCasesTrendChart/configuration.js]: @DiscoElevator needs further investigation, code was moved from: https://github.com/reportportal/service-ui/blob/master/src/main/resources/public/js/src/newWidgets/_C3ChartWidgetView.js#L66

**Comment by norbert-csaba-herczeg-epam** [comment]
@DiscoElevator If the magic number part is fine for now as-is the widgets are up-to-date with current data API-s, and your other suggestions have been addressed.

File: app/src/components/widgets/charts/failedCasesTrendChart/failedCasesTrendChart.stories.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (DiscoElevator) [reply app/src/components/widgets/charts/failedCasesTrendChart/failedCasesTrendChart.stories.jsx]: Try to add this import to `.storybook/config.js`

[the remark this answers is not in the document]
Comment (norbert-csaba-herczeg-epam) [reply app/src/components/widgets/charts/failedCasesTrendChart/failedCasesTrendChart.stories.jsx]: @DiscoElevator It worked, thanks for the tip ;-)
